src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/UnicodeHelpers.cs

```diff
@@ -29,13 +29,19 @@ internal unsafe static class UnicodeHelpers
 
         /// <summary>
         /// Helper method which creates a bitmap of all characters which are
-        /// defined per version 7.0.0 of the Unicode specification.
+        /// defined per the Unicode specification.
         /// </summary>
         [MethodImpl(MethodImplOptions.NoInlining)]
         private static uint[] CreateDefinedCharacterBitmap()
         {
             // The stream should be exactly 8KB in size.
-            var stream = typeof(UnicodeHelpers).GetTypeInfo().Assembly.GetManifestResourceStream("System.Text.Encodings.Web.Resources.unicode-8.0.0-defined-characters.bin");
+            var stream = typeof(UnicodeRange).GetTypeInfo().Assembly.GetManifestResourceStream("System.Text.Encodings.Web.Resources.unicode8definedcharacters.bin");
+
+            if (stream == null)
+            {
+                throw new BadImageFormatException();
```

Comment:
I just copied what System.Text.Encodings.CodePages does with its data resource stream, i.e. it throws an exception.

Moreover, I am not sure we actually want to fail fast here in the first place. Thoughts?